src/deprecated.jl

```diff
@@ -599,8 +599,14 @@ end
         return missing
     else
         @inbounds pool = source.pools[col]
-        return get🐱(pool, str::Tuple{Ptr{UInt8}, Int})
-    end
+        p, len = str::Tuple{Ptr{UInt8}, Int}
+        level = unsafe_string(p, len)
+        i = get(pool, level, nothing)
+        if i === nothing
+            i = get!(pool, level)
+            issorted(levels(pool)) || levels!(pool, sort(levels(pool)))
```

Comment:
Should we just sort them after insert? I can't imagine that levels will come in sorted very often, so it would avoid the extra `issorted` check. I wonder if we should also consider a sorting algorithm optimized for "nearly sorted"; probably not a big deal, but I know some algorithms have a toxic path when the input is almost sorted.

Reply:
CategoricalArrays need to do some work in `levels!` which is likely to be much more costly than calling `issorted`, so I think it's more efficient that way. If we want to improve performance here, we could need to add an argument to the function which adds levels, and have `CategoricalPool` store a Boolean to remember whether levels are currently sorted.

Anyway, the number of levels is supposed to be much lower than the number of rows (that's the whole point of CategoricalArrays), so this shouldn't really matter.